### DISCUSSION TITLE
SingleLineTextLayout example

### DISCUSSION THREAD

**Discussion by dan-willett** [Hopding/pdf-lib#1333] [Q&A]
Could someone please provide an example of using SingleLineTextLayout?  I really need to right align some text, but it would be nice to have an example of how to use this to accomplish that.

https://pdf-lib.js.org/docs/api/interfaces/singlelinetextlayout

**Comment by imsam** [comment]
Not an answer, just a +1 to this question.

I've been through the docs and can see that TextAlignment is an option in [LayoutSinglelineText](https://pdf-lib.js.org/docs/api/index#const-layoutsinglelinetext) and [LayoutMultilineText](https://pdf-lib.js.org/docs/api/index#const-layoutmultilinetext), but I can't figure out how to actually use those global functions.

I'll post here if I figure it out, but where these global functions work differently to page.drawText(), it appears to require options that I haven't needed to specify elsewhere (alignment, bounds, font)

**Reply by imsam** [reply]
Although it doesn't answer the original question, I wrote a little helper function to right-align. Not the prettiest, and it requires some manual calculation/trial-and-error with distances, but it works.

``` ts
const rightAlign = async (
	doc: PDFDocument,
	page: PDFPage,
	currentPosition: { x: number; y: number },
	moveRight: number,
	text: string,
	textOptions: PDFPageDrawTextOptions
) => {
	const helveticaFont = await doc.embedFont(StandardFonts.Helvetica)
	const textWidth = helveticaFont.widthOfTextAtSize(text, textOptions.size)

	page.moveTo(currentPosition.x, currentPosition.y)
	page.moveRight(moveRight)
	page.moveLeft(textWidth)
	page.drawText(text, textOptions)
	page.moveRight(textWidth)
	page.moveLeft(moveRight)
}
```

Calling the function would look something like this:

``` ts
rightAlign(
	pdfDoc,
	currentPage,
	currentPage.getPosition(),
	286,
	'Right-aligned text!!!',
	{
		size: 12,
		color: rgb(0.6, 0.6, 0.6),
	}
)
```
